### ISSUE TITLE
enhancement: support underbar prefix indicating known unused variable

### ISSUE THREAD

**Issue by smoser** [koalaman/shellcheck#1193]
#### For bugs
- Rule Id: SC2034
- My shellcheck version: online (0.4.6 locally)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
# shellcheck disable=2162
read up _idle </proc/uptime
echo "system up $up"

```

#### Here's what shellcheck currently says:
    $ shellcheck myscript
     
    Line 3:
    read up _idle </proc/uptime
           ^-- SC2034: _idle appears unused. Verify it or export it.

#### Here's what I wanted or expected to see:
Naming the variable with a '_' prefix in many linters is a way of acknowledging it is known as unused.  As in this example, giving it the name _idle indicates what the second field is, and thus provides value to the developer.

Using a single `_` does work.

**Comment by Dmole** [comment]
I think the current behaviour is more correct, you should use;

up=$(cut -f 1 -d " " /proc/uptime)

**Comment by smoser** [comment]
Pylint allows that by default.

**Comment by smoser** [comment]
That 'cut' is several orders of magnitude slower and relies on an external program.

**Comment by matthewpersico** [comment]
If it's not a shell convention, I wouldn't add it as a default pattern to allow. This, however, is a good example of allowing for an `enable` behavior. At the top his script, @smoser could put 

```# shellecheck enable=SCe1001```

which would negate SC2034 for _ leading vars. And if it's right under the shebang, it applies to the whole script, right?

**Comment by Dmole** [comment]
Then use

`read -d " " up < /proc/uptime`

**Comment by smoser** [comment]
`-d` argument to `read` is not posix sh.


Thanks for the suggestions of other workarounds.  I'm not interested littering my code with "# shellcheck" comments locally, or globally disabling the other-wise useful warning. 

I've seen the convention of '_' prefix to skip such unused-variables before, it provides a way to acknowledge that the variable is unused, but still provide the reader with information.  I guess others here just don't think so.

Thanks for your work on shellcheck, it really is a useful tool.

**Comment by koalaman** [comment]
I definitely see the utility, but this is the first time I've heard of this convention so I'm hesitant to adopt it. Let's revisit when it becomes more popular.

**Comment by speculatrix** [comment]
@koalaman I think it would be good to revisit this. If you're running a command which spits out more than one parameter and you don't want to pass it through awk or cut when reading the params, being able to ignore them would be useful by prepending _.

**Comment by Muzer** [comment]
I would also like you to revisit this. I've just had a similar issue, also with a `read` command. This command spits out variables which frequently go unused. So as a result, it's a huge missing feature not to have a way of marking a variable as being known to be unused.

**Comment by TinCanTech** [comment]
I use `$foo_UNUSED`, so when  `shellcheck` warns me I can immediately discard the warning.

**Comment by RomanValov** [comment]
> As a workaround for now I've done the following:
> 
> `: "$_unused" # Keep shellcheck happy`
> 

@Muzer nice hint!
